swift-sdk/Internal/EmbeddedMessagingManager.swift

```diff
@@ -133,5 +148,5 @@ class EmbeddedMessagingManager: NSObject, IterableEmbeddedMessagingManagerProtoc
     
     private var messages: [IterableEmbeddedMessage] = []
     
-    private var listeners: [String] = [] //change to protocol
+    private var listeners: [IterableEmbeddedMessagingUpdateDelegate] = []
 }
```

Comment:
hm interesting so all the logic is passed to the processor and the manager just directs calls basically.

Reply:
More or less, it's about keeping the details separate from the manager.